src/main/java/racingcar/domain/Cars.java

```diff
@@ -0,0 +1,38 @@
+package racingcar.domain;
+
+import java.util.List;
+import java.util.Random;
+
+public class Cars {
+
+    public static final int RANDOM_BOUND = 10;
+    private final List<Car> cars;
+
+    public Cars(List<Car> cars) {
+        this.cars = cars;
+    }
+
+    public boolean isEmpty() {
+        return cars == null || cars.isEmpty();
+    }
+
+    public void race(Random random) {
+        if (cars == null || cars.isEmpty()) {
+            throw new IllegalStateException("경주를 하기 위해서는 적어도 한 대의 자동차가 있어야 합니다.");
+        }
```

Comment:
생성자나 스태틱 팩토리 메서드에서 검증하면 어떨까요? 
사실 지금 예외처리하신 이 경우에는 `Cars` 라는 객체의 존재 자체가 묘연해지는 상황이기때문에, 그 단에서 막는게 좋을 것 같습니다.

Reply:
원래 목적은 NullPointerException을 막기 위함이었는데요. 주신 의견을 수렴하기 위해서는 아래 처럼 cars 인스턴스 변수에 new ArrayList()를 처음부터 할당하는 방식을 택해야할것 같습니다.

```suggestion
public class Cars {

    public static final int RANDOM_BOUND = 10;
    private final List<Car> cars = new ArrayList<>(); // 처음부터 초기화

    public Cars(List<Car> cars) {
        this.cars.addAll(cars); // 할당하는 방법이 아닌 추가 하는 방법으로
    }

    public boolean isEmpty() {
        return cars == null || cars.isEmpty();
    }

    public void race(Random random) {
        for (Car car : cars) {
            car.moveOrStop(random.nextInt(RANDOM_BOUND));
        }
    }
```

Reply:
스태틱 팩토리 메서드에서 이미 처리되고 있어서, 기존 검증 부분은 삭제하는 방식으로 수정했습니다.